core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,157 @@
+
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for tags of exploration"""
+
+from __future__ import annotations
+
+from core.domain import exp_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+from typing import List
+
+(exp_models, ) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class GetNumberOfExpHavingInvalidTagsListJob(base_jobs.JobBase):
+    """Job that returns invalid exploration tags"""
+
+    def _check_tags_validity(self, tags: List[str]) -> bool:
+        """checks whether tags list is valid or not.
+
+        Args:
+            tags: List[str]. The list of all tags in exploration.
+
+        Returns:
+            bool. If the list containing tags have length more than 10 or
+            individual tag have length more than 30, then it returns True
+            otherwise False.
+        """
+        if len(tags) > 10:
+            return True
+        if len(set(tags)) < len(tags):
+            return True
+        for tag in tags:
+            if tag.strip() == '':
+                return True
+            elif len(tag) > 30:
+                return True
+        return False
+
+    def _get_description_of_wrong_exp(self, tags: List[str]) -> str:
```

Comment:
Here you are basically duplicating what you did above, maybe you can rather have one function that will raise errors or that will return either `result.Ok` or `result.Err` (from the `result` library).